diff --git a/src/app.vala b/src/app.vala
index 9bb6a90..e08d177 100644
--- a/src/app.vala
+++ b/src/app.vala
@@ -225,10 +225,6 @@ namespace GameHub
 
 		public static int main(string[] args)
 		{
-			#if MANETTE
-			X.init_threads();
-			#endif
-
 			var app = new Application();
 
 			Utils.Logger.init();
diff --git a/src/meson.build b/src/meson.build
index 2acdeac..c8b79f7 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -256,9 +256,6 @@ manette = dependency('manette-0.2', required: false)
 if manette.found()
 	add_global_arguments('-D', 'MANETTE', language: 'vala')
 	deps += manette
-	deps += dependency('x11')
-	deps += dependency('gdk-x11-3.0')
-	deps += dependency('xtst')
 	sources += 'utils/Gamepad.vala'
 	sources += 'ui/dialogs/SettingsDialog/pages/general/Controller.vala'
 endif
diff --git a/src/utils/Gamepad.vala b/src/utils/Gamepad.vala
index 95dd243..ffaef53 100644
--- a/src/utils/Gamepad.vala
+++ b/src/utils/Gamepad.vala
@@ -254,16 +254,5 @@ namespace GameHub.Utils.Gamepad
 	// hack, but works (on X11)
 	private static void emit_key_event(uint key_code, bool press)
 	{
-		if(key_code == 0) return;
-		foreach(var wnd in Gtk.Window.list_toplevels())
-		{
-			if(wnd.is_active)
-			{
-				unowned X.Display xdisplay = (wnd.screen.get_display() as Gdk.X11.Display).get_xdisplay();
-				XTest.fake_key_event(xdisplay, xdisplay.keysym_to_keycode((ulong) key_code), press, X.CURRENT_TIME);
-				Gamepad.ButtonPressed = true;
-				break;
-			}
-		}
 	}
 }
