From 1b189aad89ac23bae361ac1c46fb26f8766e685e Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Fri, 4 Feb 2022 19:14:24 +0100
Subject: [PATCH 1/4] fix wayland

---
 cmake/Modules/FindWayland.cmake | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/cmake/Modules/FindWayland.cmake b/cmake/Modules/FindWayland.cmake
index 31983a8a7..0fd8260c8 100644
--- a/cmake/Modules/FindWayland.cmake
+++ b/cmake/Modules/FindWayland.cmake
@@ -25,8 +25,6 @@ find_package(PkgConfig)
 pkg_check_modules(PKG_WAYLAND QUIET wayland-client wayland-server wayland-egl
                   wayland-cursor)
 
-set(WAYLAND_DEFINITIONS ${PKG_WAYLAND_CFLAGS})
-
 find_path(
   WAYLAND_CLIENT_INCLUDE_DIRS
   NAMES wayland-client.h
@@ -125,8 +123,8 @@ foreach(component "Client" "Server" "EGL" "Cursor")
                    "${WAYLAND_${component_u}_INCLUDE_DIRS}")
 
       set_target_properties(
-        Wayland::${component} PROPERTIES INTERFACE_COMPILE_DEFINITIONS
-                                         "${WAYLAND_DEFINITIONS}")
+          Wayland::${component} PROPERTIES INTERFACE_COMPILE_OPTIONS
+                                         "${PKG_WAYLAND_CFLAGS}")
     endif()
   endif()
 endforeach()
-- 
2.35.1

