From 707c3d106456e2a9ab5ba0965be38ef3eb9d2155 Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Fri, 4 Feb 2022 20:22:16 +0100
Subject: [PATCH 2/3] disable browser and vst

---
 plugins/CMakeLists.txt | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/plugins/CMakeLists.txt b/plugins/CMakeLists.txt
index b9f2924df..2af66df7f 100644
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -5,18 +5,22 @@ if(NOT ENABLE_PLUGINS)
 endif()
 
 function(check_obs_browser)
-  if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt)
-    add_subdirectory(obs-browser)
-  else()
-    obs_status(FATAL_ERROR "obs-browser submodule not available.")
+  if(ENABLE_BROWSER_SOURCE)
+    if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt)
+      add_subdirectory(obs-browser)
+    else()
+      obs_status(FATAL_ERROR "obs-browser submodule not available.")
+    endif()
   endif()
 endfunction()
 
 function(check_obs_vst)
-  if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/obs-vst/CMakeLists.txt)
-    add_subdirectory(obs-vst)
-  else()
-    obs_status(FATAL_ERROR "obs-vst submodule not available.")
+  if(ENABLE_VST)
+    if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/obs-vst/CMakeLists.txt)
+      add_subdirectory(obs-vst)
+    else()
+      obs_status(FATAL_ERROR "obs-vst submodule not available.")
+    endif()
   endif()
 endfunction()
 
-- 
2.35.1

