From c43438b3a4236c982c69e1d5eca471b2ac1a21f7 Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Fri, 4 Feb 2022 20:52:08 +0100
Subject: [PATCH 3/3] fix rtmp

---
 plugins/obs-outputs/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/obs-outputs/CMakeLists.txt b/plugins/obs-outputs/CMakeLists.txt
index cde40922f..f4626de0f 100644
--- a/plugins/obs-outputs/CMakeLists.txt
+++ b/plugins/obs-outputs/CMakeLists.txt
@@ -47,11 +47,11 @@ if(OS_WINDOWS)
   target_link_libraries(obs-outputs PRIVATE ws2_32 winmm Iphlpapi)
 endif()
 
-if(ENABLE_RTMPS STREQUAL "AUTO" OR ENABLE_RTMPS STREQUAL "ON")
+if(ENABLE_RTMPS)
   find_package(MbedTLS)
   find_package(ZLIB)
   if(NOT MBEDTLS_FOUND OR NOT ZLIB_FOUND)
-    if(ENABLE_RTMPS STREQUAL "ON")
+    if(ENABLE_RTMPS)
       obs_status(FATAL_ERROR
                  "mbedTLS or zlib not found, but required for RTMPS support.")
       return()
-- 
2.35.1

