From f41bbb957d14d87306bfd7db23a5c5f61df6a80f Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Fri, 4 Feb 2022 20:26:53 +0100
Subject: [PATCH 4/4] fix build without scripting

---
 cmake/Modules/ObsHelpers.cmake | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/cmake/Modules/ObsHelpers.cmake b/cmake/Modules/ObsHelpers.cmake
index b19608241..5f09877ce 100644
--- a/cmake/Modules/ObsHelpers.cmake
+++ b/cmake/Modules/ObsHelpers.cmake
@@ -140,9 +140,11 @@ function(setup_obs_app target)
   get_property(OBS_MODULE_LIST GLOBAL PROPERTY OBS_MODULE_LIST)
   add_dependencies(${target} ${OBS_MODULE_LIST})
 
-  get_property(OBS_SCRIPTING_MODULE_LIST GLOBAL
-               PROPERTY OBS_SCRIPTING_MODULE_LIST)
-  add_dependencies(${target} ${OBS_SCRIPTING_MODULE_LIST})
+  if(ENABLE_SCRIPTING)
+    get_property(OBS_SCRIPTING_MODULE_LIST GLOBAL
+                 PROPERTY OBS_SCRIPTING_MODULE_LIST)
+    add_dependencies(${target} ${OBS_SCRIPTING_MODULE_LIST})
+  endif()
 
   # detect outdated obs-browser submodule
   if(NOT TARGET OBS::browser AND TARGET obs-browser)
-- 
2.35.1

