diff --git a/comm/mail/installer/package-manifest.in b/comm/mail/installer/package-manifest.in.a
index 89e11cd..0be6981 100644
--- a/comm/mail/installer/package-manifest.in
+++ b/comm/mail/installer/package-manifest.in.a
@@ -121,13 +121,6 @@
 @BINPATH@/api-ms-win-*.dll
 @BINPATH@/ucrtbase.dll
 #endif
-#endif
-#ifdef MOZ_GTK
-@BINPATH@/@DLL_PREFIX@mozgtk@DLL_SUFFIX@
-#ifdef MOZ_WAYLAND
-@BINPATH@/@DLL_PREFIX@mozwayland@DLL_SUFFIX@
-#endif
-#endif
 
 ; We don't have a complete view of which dlls to expect when doing an artifact
 ; build because we haven't run the relevant parts of configure, so we guess
