diff --git a/CMakeLists.txt b/CMakeLists.txt.a
index 022f068..d0fb82d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt.a
@@ -28,6 +28,7 @@ include(ECMPoQmTools)
 
 set(EXCLUDE_DEPRECATED_BEFORE_AND_AT 0 CACHE STRING "Control the range of deprecated API excluded from the build [default=0].")
 
+option(HAVE_X11 "Use X11" OFF)
 option(BUILD_QCH "Build API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)" OFF)
 add_feature_info(QCH ${BUILD_QCH} "API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)")
 
@@ -48,7 +49,7 @@ find_package(KF5DBusAddons ${KF_DEP_VERSION} REQUIRED)
 find_package(KF5WindowSystem ${KF_DEP_VERSION} REQUIRED)
 
 # no X11 stuff on mac
-if (NOT APPLE)
+if (HAVE_X11)
     find_package(XCB MODULE COMPONENTS XCB KEYSYMS XKB OPTIONAL_COMPONENTS XTEST)
     set_package_properties(XCB PROPERTIES DESCRIPTION "X protocol C-language Binding"
                        URL "http://xcb.freedesktop.org"
@@ -56,14 +57,9 @@ if (NOT APPLE)
                       )
 
     find_package(X11)
-endif()
-
-set(HAVE_X11 0)
-
-if(X11_FOUND AND XCB_XCB_FOUND)
-    set(HAVE_X11 1)
     find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED X11Extras)
 endif()
+
 add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050f02)
 add_definitions(-DKF_DISABLE_DEPRECATED_BEFORE_AND_AT=0x055100)
 # Subdirectories
